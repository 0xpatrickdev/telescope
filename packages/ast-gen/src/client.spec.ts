import {
    createClient
} from './client';
import generate from '@babel/generator';

import { mutations } from './__fixtures__';

const expectCode = (ast) => {
    expect(
        generate(ast).code
    ).toMatchSnapshot();
}
const printCode = (ast) => {
    console.log(
        generate(ast).code
    );
}

it('createClient', async () => {
    printCode(createClient({
        name: 'getSigningOsmosisClient',
        packages: [
            'osmosis.gamm.v1beta1',
            'osmosis.superfluid.v1beta1',
            'osmosis.lockup'
        ]
    }));
});