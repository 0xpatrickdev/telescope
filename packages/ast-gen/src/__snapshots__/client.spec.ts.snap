// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createClient 1`] = `
"export const getSigningOsmosisClient = async ({
  rpcEndpoint,
  signer
}: {
  rpcEndpoint: string;
}) => {
  const registry = new Registry(defaultRegistryTypes);
};"
`;
